File: 15_Task(Sort)/HomeWork/src/Sort/BubbleSort.java
package Sort;

import java.util.Comparator;

import Collection.Collection;

public class BubbleSort<T> extends Sort<T> {

  public BubbleSort(Comparator<T> comparator) {
    super(comparator);
  }

  @Override
  public void sort(Collection<T> collection) {
    int n = collection.count() - 1;
    T temp;

    for (int i = 0; i <= n; i++) {
      for (int j = 0; j < n-i; j++) {
        if (comparator.compare(collection.get(j), collection.get(j+1)) > 0) {
          temp = collection.get(j);
          collection.set(j, collection.get(j+1));
          collection.set(j+1, temp);
        }
      }
    }

  }
  
}

File: 15_Task(Sort)/DP_SortCollections/DP_SortCollections/BubbleSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DP_SortCollections
{
    class BubbleSort<T> : SortStrategy<T>
    {
        public override void Sort(ISortable<T> container)
        {
            if (Comparer != null)
            {
                for (int i = 0; i < container.Length; i++)
                {
                    for (int k = 0; k < container.Length - 1; k++)
                    {
                        if (Comparer.Compare(container[k], container[k + 1]) == 1)
                        {
                            T temp = container[k + 1];
                            container[k + 1] = container[k];
                            container[k] = temp;
                        }
                    }
                }
            }
        }
    }
}
